### PR TITLE
add group_size/pool_size for lmp and fp

### PR DESCRIPTION
**Pull request by Angel-Jia**
Adding group_size/pool_size for `lmp` and `fp`. A node can run up to `group_size` tasks and at most `pool_size` tasks are running at the same time.

### CODE REVIEW

File: dpgen2/superop/prep_run_fp.py [hunk 1]
```diff
@@ -125,6 +125,8 @@ def _prep_run_fp(
     run_template_config = run_config.pop("template_config")
     prep_executor = init_executor(prep_config.pop("executor"))
     run_executor = init_executor(run_config.pop("executor"))
+    group_size = run_config.pop("group_size") if "group_size" in run_config else None
+    pool_size = run_config.pop("pool_size") if "pool_size" in run_config else None
```
Comment (wanghan-iapcm) [inline dpgen2/superop/prep_run_fp.py:129]: I would recommend put the two parameters to a section named `template_slice_config`, and pass the dictionary as extra kwargs to the class `Slices`. Then **any** extra parameters to the `Slices` is supported.

File: dpgen2/superop/prep_run_lmp.py [hunk 2]
```diff
@@ -129,6 +129,8 @@ def _prep_run_lmp(
     run_template_config = run_config.pop("template_config")
     prep_executor = init_executor(prep_config.pop("executor"))
     run_executor = init_executor(run_config.pop("executor"))
+    group_size = run_config.pop("group_size") if "group_size" in run_config else None
+    pool_size = run_config.pop("pool_size") if "pool_size" in run_config else None
```
Comment (wanghan-iapcm) [inline dpgen2/superop/prep_run_lmp.py:133]: The same comment as `run_fp`

**Review by wanghan-iapcm — CHANGES_REQUESTED** [review]

File: dpgen2/superop/prep_run_fp.py [hunk 3]
```diff
@@ -156,6 +163,8 @@ def _prep_run_fp(
                 input_parameter=["task_name"],
                 input_artifact=["task_path"],
                 output_artifact=["log", "labeled_data"],
+                pool_size=pool_size,
+                group_size=group_size,
```
Comment (wanghan-iapcm) [inline dpgen2/superop/prep_run_fp.py:167]: The template slice config should be passed here

File: dpgen2/superop/prep_run_fp.py [hunk 4]
```diff
@@ -125,6 +125,13 @@ def _prep_run_fp(
     run_template_config = run_config.pop("template_config")
     prep_executor = init_executor(prep_config.pop("executor"))
     run_executor = init_executor(run_config.pop("executor"))
+    template_slice_config = run_config.pop("template_slice_config", None)
+    group_size = (
+        template_slice_config.get("group_size", None) if template_slice_config else None
+    )
+    pool_size = (
+        template_slice_config.get("pool_size", None) if template_slice_config else None
+    )
```
Comment (wanghan-iapcm) [inline dpgen2/superop/prep_run_fp.py:134]: The two keys are not necessary when the `template_slice_config` is correctly implemented.

**Review by wanghan-iapcm — CHANGES_REQUESTED** [review]

File: dpgen2/superop/prep_run_lmp.py [hunk 5]
```diff
@@ -129,6 +129,7 @@ def _prep_run_lmp(
     run_template_config = run_config.pop("template_config")
     prep_executor = init_executor(prep_config.pop("executor"))
     run_executor = init_executor(run_config.pop("executor"))
+    template_slice_config = run_config.pop("template_slice_config", {})
```
Comment (wanghan-iapcm) [inline dpgen2/superop/prep_run_lmp.py:132]: Please add the argument definition for the `template_slice_config` to the [`step_config`](https://github.com/deepmodeling/dpgen2/blob/master/dpgen2/utils/step_config.py). 
An example on the `template_config` can be found [here](https://github.com/deepmodeling/dpgen2/blob/89f7c9590a8a9f914b0122a379448d59ad0ab0ca/dpgen2/utils/step_config.py#L106-L113) and [here](https://github.com/deepmodeling/dpgen2/blob/89f7c9590a8a9f914b0122a379448d59ad0ab0ca/dpgen2/utils/step_config.py#L68-L94)

**Review by wanghan-iapcm — CHANGES_REQUESTED** [review]

File: dpgen2/utils/step_config.py [hunk 6]
```diff
@@ -101,6 +111,13 @@ def step_conf_args():
             default={"image": default_image},
             doc=doc_template,
         ),
+        Argument(
+            "template_slice_config",
+            dict,
+            template_slice_conf_args(),
+            optional=True,
+            doc=doc_template_slice,
```
Comment (wanghan-iapcm) [inline dpgen2/utils/step_config.py:119]:
```suggestion
            default=None,
            doc=doc_template_slice,
```

File: dpgen2/superop/prep_run_lmp.py (hunk 5 above)
Comment (wanghan-iapcm) [inline dpgen2/superop/prep_run_lmp.py:132]:
```suggestion
    template_slice_config = run_config.pop("template_slice_config")
```

File: dpgen2/superop/prep_run_fp.py [hunk 7]
```diff
@@ -125,6 +125,7 @@ def _prep_run_fp(
     run_template_config = run_config.pop("template_config")
     prep_executor = init_executor(prep_config.pop("executor"))
     run_executor = init_executor(run_config.pop("executor"))
+    template_slice_config = run_config.pop("template_slice_config", {})
```
Comment (wanghan-iapcm) [inline dpgen2/superop/prep_run_fp.py:128]:
```suggestion
    template_slice_config = run_config.pop("template_slice_config")
```

**Review by wanghan-iapcm — CHANGES_REQUESTED** [review]

File: dpgen2/utils/step_config.py (hunk 6 above)
Comment (Angel-Jia) [reply dpgen2/utils/step_config.py:119]: @wanghan-iapcm Every step config will have a `template_slice_config` if set `default=None` and a TypeError will be raised:
```
dpgen2/superop/prep_run_dp_train.py", line 133, in _prep_run_dp_train
    prep_train = Step(
TypeError: Step.__init__() got an unexpected keyword argument 'template_slice_config'
```

Comment (wanghan-iapcm) [reply dpgen2/utils/step_config.py:119]:
```suggestion
            default={},
            doc=doc_template_slice,
```
This one?

**Review by wanghan-iapcm — APPROVED** [review]

**Pull request merged**